src/XTMF2.Web.Server/Controllers/ModelSystemEditorController.cs

```diff
@@ -43,9 +41,48 @@ public ModelSystemEditorController(XTMFRuntime runtime)
         /// <param name="project"></param>
         /// <param name="name"></param>
         /// <returns></returns>
-        [HttpGet]
-        public ActionResult Get()
-        {
+        [ProducesResponseType(StatusCodes.Status422UnprocessableEntity)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [HttpPost("projects/{projectName}/model-systems/{modelSystemName}/open-session")]
+        public ActionResult OpenSession(string projectName, string modelSystemName, [FromServices] User user) {
+            string error = default;
+            if(!Utils.XtmfUtils.GetModelSystemHeader(_xtmfRuntime,user,projectName,modelSystemName,out var modelSystemHeader, ref error)) {
+                return new NotFoundObjectResult(error);
+            }
+            if(!Utils.XtmfUtils.GetProjectSession(_xtmfRuntime, user, projectName, out var projectSession, ref error)) {
+                return new NotFoundObjectResult(error);
+            }
+            if(!projectSession.EditModelSystem(user,modelSystemHeader,out var session, ref error)) {
+                return new UnprocessableEntityObjectResult(error);
+            }
+            return new OkResult();
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="projectName"></param>
+        /// <param name="modelSystemName"></param>
+        /// <param name="user"></param>
+        /// <returns></returns>
+        [ProducesResponseType(StatusCodes.Status422UnprocessableEntity)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [HttpPost("projects/{projectName}/model-systems/{modelSystemName}/end-session")]
+        public ActionResult EndSession(string projectName, string modelSystemName, [FromServices] User user) {
+            string error = default;
+            if(!Utils.XtmfUtils.GetModelSystemHeader(_xtmfRuntime,user,projectName,modelSystemName,out var modelSystemHeader, ref error)) {
+                return new NotFoundObjectResult(error);
+            }
+            if(!Utils.XtmfUtils.GetProjectSession(_xtmfRuntime, user, projectName, out var projectSession, ref error)) {
+                return new NotFoundObjectResult(error);
+            }
+            if(!projectSession.EditModelSystem(user,modelSystemHeader,out var session, ref error)) {
+                return new UnprocessableEntityObjectResult(error);
+            }
+            //dispose of the session
+            session.Dispose();
```

Comment:
This is going to cause a leak for both projectSession and session since disposing it will only remove the newly created session.